### DISCUSSION TITLE
An issue with nested eager loading

### DISCUSSION THREAD

**Discussion by nbayramberdiyev** [larastan/larastan#2039] [Q&A] [answered]
After upgrading Laravel to the latest version (11.23.2), I encountered the following error when I eager load a nested relationship by providing a nested array to the `with` method:

```
Parameter #1 $relations of method Illuminate\Database\Eloquent\Builder<App\Models\Entry\Entry>::with() 
 expects array<(Closure(Illuminate\Database\Eloquent\Relations\Relation<*, *, *>): mixed)|string>|string, 
 array{0: 'contractor:id,name', campaign:id,partner_id,name: array{'partner:id,name'}} given.
```

Here's the query:
```php
->with([
    'contractor:id,name',
    'campaign:id,partner_id,name' => ['partner:id,name'],
])
```

This should be possible, according to the [documentation](https://laravel.com/docs/11.x/eloquent-relationships#nested-eager-loading). The error disappears if I load it this way:

```php
->with([
    'contractor:id,name',
    'campaign.partner',
])
```
However, I'd like to retrieve specific columns for performance and memory reasons.

My `phpstan.neon` file looks like this:
```neon
includes:
    - ./vendor/larastan/larastan/extension.neon

parameters:
    ignoreErrors:
        - identifier: missingType.generics

    paths:
        - app
        - config
        - database
        - tests

    level: 8
```

Laravel: v11.23.2
Larastan: v2.9.8

What could be causing this issue?

**Comment by calebdw** [comment]
Hello,

The docs for that method were updated in v11.23.0 and unfortunately recursive array definitions are not possible yet. This will be fixed over there---nothing for Larastan to do here.

Thanks!

**Reply by nbayramberdiyev** [reply]
Thanks for the response. I can't seem to find the updated documentation. Would you be able to share a link?

**Reply by calebdw** [reply, marked as answer]
This was fixed in https://github.com/laravel/framework/pull/52762
